src/proto/single/mod.rs

```diff
@@ -66,8 +70,8 @@ const JOB_DEFAULT_BACKTRACE: usize = 0;
 )]
 pub struct Job {
     /// The job's unique identifier.
-    #[builder(default = "utils::gen_random_jid()")]
-    pub(crate) jid: String,
+    #[builder(default = "JobId::random()")]
+    pub jid: JobId,
```

Comment:
I guess not only `args` (#26 ), but also `jid` and `kind` can be public on Job.
![image](https://github.com/jonhoo/faktory-rs/assets/117771945/7c8c69ad-b9cd-4ef5-91a8-b672d2f7d0ca)

Reply:
The reason `jid` wasn't public was so that folks make sure to construct the field through our constructors rather than modifying it manually. I think that's probably still a good idea.

The reason `args` wasn't public was so that we could change how they're stored in the future if we want to (e.g., move away from `serde_json::Value` or use something other than a `Vec`). Not sure how likely that is, but may still be worthwhile?